sequence-diagrams/backend/create-birth-registration-field-agent.md

```diff
@@ -0,0 +1,100 @@
+
+sequenceDiagram
+    autonumber
+    participant GraphQL gateway
+    participant Workflow
+    participant User management
+    participant Hearth
+    participant Config
+    participant Metrics
+    participant Notifications
+    participant Influx DB
+    participant Search
+    participant ElasticSearch
+    participant Documents
+    participant Minio
+    participant Country-Config
+
+    GraphQL gateway->>Workflow: POST /create-record
+    Workflow--)Hearth: Fetch task from identifier(draft-id)
+    Note over Workflow,Hearth: Check existing draft-IDs. If found, return previous declaration info to gateway.
+
+    Workflow--)Search: Search for duplicates
+
+    Workflow--)Documents: POST certificate details to /upload
+    Documents->>Minio: Upload certificate documents
+
+    Workflow--)Country-Config: GET trackingId from /tracking-id
+
+    Workflow->>User management: Fetch user/system information
+    Workflow->>Hearth: Get practitioner resource
+
+    loop PractitionerRole Locations
+      Workflow->>Hearth: Get location by user's practitionerId
+    end
+    Note over Workflow,Hearth: Update bundle with practitioner details
+
+    Workflow->>Hearth: Save bundle
+    Note over Workflow,Hearth: Get hearth response for all entries
+
+    Note over Workflow: Merge changed resources<br /> into record with <br /> hearth's response bundle
+
+    Workflow--)Metrics: POST bundle to /events/{event}/sent-notification
+    Metrics->>Influx DB: Write user audit point "IN_PROGRESS"
+
+    loop location levels 4, 3, 2
+        Metrics->>Workflow: Get parent of Location
+        Workflow->>Hearth: Get parent of Location
+    end
+    Note over Metrics,Hearth: Generate time logged point
+
+    loop location levels 4, 3, 2
+        Metrics->>Workflow: Get parent of Location
+        Workflow->>Hearth: Get parent of Location
+    end
+    Note over Metrics,Hearth: Generate declaration started point
+    Metrics->>Influx DB: Write audit points
+
+    Note over Workflow,Search: For duplicate declarations, starts here
+    Workflow--)Search: Index bundle
+
+    %% upsertEvent
+    Search->>ElasticSearch: Search by composition id
+    Note over Search,ElasticSearch: Get operation history and createdAt
+
+    %% createIndexBody
+      %% createChildIndex
+      %% addEventLocation
+    Search->>Hearth: Get event location for creating child index
+
+    %% createDeclarationIndex
+    Search->>Hearth: Get declarationJurisdictionIds for declaration index
+    Search->>ElasticSearch: Get createdBy
+
+    %% createStatusHistory
+    Search->>User management: Get user for status history
+    Note over Search,Hearth: Compose new history entry
+
+    Search->>ElasticSearch: Index composition
+    Workflow->>Hearth: Save bundle to Hearth
+    Note over Workflow,Search: For duplicate declarations, ends here
+
+    Workflow--)Config: Check if notification is enabled
+    Workflow--)Search: Index bundle
+
+    Workflow--)Notifications: Send notification if enabled
```

Comment:
```suggestion
    Workflow--)Config: Check if notification is enabled
    Workflow--)Notifications: Send notification if enabled

    Workflow--)Search: Index bundle
```

Reply:
We should also mention how search services fetches data from hearth for populating location details

Reply:
The order of the flow is currently how it is demonstrated here. We could update the flow to serialize the notification flow.